tls/s2n_connection_serialize.c

```diff
@@ -119,6 +134,19 @@ int s2n_connection_serialize(struct s2n_connection *conn, uint8_t *buffer, uint3
         POSIX_GUARD_RESULT(s2n_connection_serialize_secrets(conn, &output));
     }
 
+    /* Users should not be able to send/recv on the connection after serialization as that
+     * could lead to nonce reuse. We close the connection to prevent the application from sending
+     * more application data. However, the application could still send a close_notify alert record
+     * to shutdown the connection, so we also intentionally wipe keys and disable encryption.
+     *
+     * A plaintext close_notify alert is not a security concern, although the peer will likely consider
+     * it an error.
+     */
+    POSIX_GUARD_RESULT(s2n_connection_set_closed(conn));
+    POSIX_GUARD_RESULT(s2n_crypto_parameters_wipe(conn->secure));
+    conn->server = conn->initial;
+    conn->client = conn->initial;
```

Comment:
If you wipe secure, do you need to switch to using initial or can you just keep using the wiped secure?